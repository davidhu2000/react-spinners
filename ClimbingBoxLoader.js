(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'emotion', 'react', 'prop-types', '@emotion/core', 'recompose/onlyUpdateForKeys'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('emotion'), require('react'), require('prop-types'), require('@emotion/core'), require('recompose/onlyUpdateForKeys'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.emotion, global.react, global.propTypes, global.core, global.onlyUpdateForKeys);
    global.ClimbingBoxLoader = mod.exports;
  }
})(this, function (exports, _emotion, _react, _propTypes, _core, _onlyUpdateForKeys) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _onlyUpdateForKeys2 = _interopRequireDefault(_onlyUpdateForKeys);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _classCallCheck(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
      throw new TypeError("Cannot call a class as a function");
    }
  }

  var _createClass = function () {
    function defineProperties(target, props) {
      for (var i = 0; i < props.length; i++) {
        var descriptor = props[i];
        descriptor.enumerable = descriptor.enumerable || false;
        descriptor.configurable = true;
        if ("value" in descriptor) descriptor.writable = true;
        Object.defineProperty(target, descriptor.key, descriptor);
      }
    }

    return function (Constructor, protoProps, staticProps) {
      if (protoProps) defineProperties(Constructor.prototype, protoProps);
      if (staticProps) defineProperties(Constructor, staticProps);
      return Constructor;
    };
  }();

  function _possibleConstructorReturn(self, call) {
    if (!self) {
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    }

    return call && (typeof call === "object" || typeof call === "function") ? call : self;
  }

  function _inherits(subClass, superClass) {
    if (typeof superClass !== "function" && superClass !== null) {
      throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
    }

    subClass.prototype = Object.create(superClass && superClass.prototype, {
      constructor: {
        value: subClass,
        enumerable: false,
        writable: true,
        configurable: true
      }
    });
    if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
  }

  var climbingBox = (0, _core.keyframes)('0%{transform:translate(0,-1em) rotate(-45deg)}5%{transform:translate(0,-1em) rotate(-50deg)}20%{transform:translate(1em,-2em) rotate(47deg)}25%{transform:translate(1em,-2em) rotate(45deg)}30%{transform:translate(1em,-2em) rotate(40deg)}45%{transform:translate(2em,-3em) rotate(137deg)}50%{transform:translate(2em,-3em) rotate(135deg)}55%{transform:translate(2em,-3em) rotate(130deg)}70%{transform:translate(3em,-4em) rotate(217deg)}75%{transform:translate(3em,-4em) rotate(220deg)}100%{transform:translate(0,-1em) rotate(-225deg)}');

  var Loader = function (_React$Component) {
    _inherits(Loader, _React$Component);

    function Loader() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, Loader);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Loader.__proto__ || Object.getPrototypeOf(Loader)).call.apply(_ref, [this].concat(args))), _this), _this.style = function () {
        var color = _this.props.color;


        return (0, _core.css)('{position:absolute;left:0;bottom:-0.1em;height:1em;width:1em;background-color:transparent;border-radius:15%;border:0.25em solid ', color, ';transform:translate(0,-1em) rotate(-45deg);animation-fill-mode:both;animation:', climbingBox, ' 2.5s infinite cubic-bezier(.79,0,.47,.97);}');
      }, _this.wrapper = function () {
        var _this$props = _this.props,
            size = _this$props.size,
            sizeUnit = _this$props.sizeUnit;


        return (0, _core.css)('{position:absolute;top:50%;left:50%;margin-top:-2.7em;margin-left:-2.7em;width:5.4em;height:5.4em;font-size:', '' + size + sizeUnit, ';}');
      }, _this.hill = function () {
        var color = _this.props.color;


        return (0, _core.css)('{position:absolute;width:7.1em;height:7.1em;top:1.7em;left:1.7em;border-left:0.25em solid ', color, ';transform:rotate(45deg);}');
      }, _this.container = function () {
        var container = (0, _core.css)('{position:relative;width:7.1em;height:7.1em;}');

        return _this.props.css ? (0, _core.css)(container, ';', _this.props.css) : container;
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(Loader, [{
      key: 'render',
      value: function render() {
        var loading = this.props.loading;


        return loading ? _react2.default.createElement(
          'div',
          {
            className: (0, _emotion.css)(this.container())
          },
          _react2.default.createElement(
            'div',
            {
              className: (0, _emotion.css)(this.wrapper())
            },
            _react2.default.createElement('div', {
              className: (0, _emotion.css)(this.style())
            }),
            _react2.default.createElement('div', {
              className: (0, _emotion.css)(this.hill())
            })
          )
        ) : null;
      }
    }]);

    return Loader;
  }(_react2.default.Component);

  Loader.propTypes = {
    loading: _propTypes2.default.bool,
    color: _propTypes2.default.string,
    size: _propTypes2.default.number,
    sizeUnit: _propTypes2.default.string,
    css: _propTypes2.default.string
  };

  Loader.defaultProps = {
    loading: true,
    color: '#000000',
    size: 15,
    sizeUnit: 'px',
    css: ''
  };

  var Component = (0, _onlyUpdateForKeys2.default)(['loading', 'color', 'size', 'sizeUnit', 'css'])(Loader);
  Component.defaultProps = Loader.defaultProps;
  exports.default = Component;
});