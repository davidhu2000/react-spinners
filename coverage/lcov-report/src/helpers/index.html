
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    