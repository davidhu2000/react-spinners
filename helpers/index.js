(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.index = mod.exports;
  }
})(this, function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  var calculateRgba = exports.calculateRgba = function calculateRgba(color, opacity) {
    if (color[0] === '#') {
      color = color.slice(1);
    }

    if (color.length === 3) {
      var res = '';
      color.split('').forEach(function (c) {
        res += c;
        res += c;
      });
      color = res;
    }

    var rgbValues = color.match(/.{2}/g).map(function (hex) {
      return parseInt(hex, 16);
    }).join(', ');
    return 'rgba(' + rgbValues + ', ' + opacity + ')';
  };
});