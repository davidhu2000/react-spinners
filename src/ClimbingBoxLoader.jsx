/** @jsx jsx */
import React from 'react';
import PropTypes from 'prop-types';
import { keyframes, css, jsx } from '@emotion/core';
import onlyUpdateForKeys from 'recompose/onlyUpdateForKeys';

const climbingBox = keyframes`
  0% {transform:translate(0, -1em) rotate(-45deg)} 
  5% {transform:translate(0, -1em) rotate(-50deg)}
  20% {transform:translate(1em, -2em) rotate(47deg)}
  25% {transform:translate(1em, -2em) rotate(45deg)}
  30% {transform:translate(1em, -2em) rotate(40deg)}
  45% {transform:translate(2em, -3em) rotate(137deg)}
  50% {transform:translate(2em, -3em) rotate(135deg)}
  55% {transform:translate(2em, -3em) rotate(130deg)}
  70% {transform:translate(3em, -4em) rotate(217deg)}
  75% {transform:translate(3em, -4em) rotate(220deg)}
  100% {transform:translate(0, -1em) rotate(-225deg)}
`;

class Loader extends React.Component {
    style = () => {
      const { color } = this.props;

      return css`{   
            position: absolute;
            left: 0;
            bottom: -0.1em;
            height: 1em;
            width: 1em;
            background-color: transparent;
            border-radius: 15%;
            border: 0.25em solid ${color};
            transform: translate(0, -1em) rotate(-45deg);
            animation-fill-mode: both;
            animation: ${climbingBox} 2.5s infinite cubic-bezier(.79, 0, .47, .97);
        }`;
    };

    wrapper = () => {
      const { size, sizeUnit } = this.props;

      return css`{        
            position: absolute;
            top: 50%;
            left: 50%;
            margin-top: -2.7em;
            margin-left: -2.7em;
            width: 5.4em;
            height: 5.4em;
            font-size: ${`${size}${sizeUnit}`};
        }`;
    };


    hill = () => {
      const { color } = this.props;

      return css`{
            position: absolute;
            width: 7.1em;
            height: 7.1em;
            top: 1.7em;
            left: 1.7em;
            border-left: 0.25em solid ${color};
            transform: rotate(45deg);
        }`;
    };

  container = () => {
    const container = css`{
            position: relative;
            width: 7.1em;
            height: 7.1em;
        }`;

    return this.props.css ? css`${container};${this.props.css}` : container;
  };

  render() {
    const { loading } = this.props;

    return loading ?
      <div css={this.container()}>
        <div css={this.wrapper()}>
          <div css={this.style()} />
          <div css={this.hill()} />
        </div>
      </div> : null;
  }
}

Loader.propTypes = {
  loading: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.number,
  sizeUnit: PropTypes.string,
  css: PropTypes.string
};

Loader.defaultProps = {
  loading: true,
  color: '#000000',
  size: 15,
  sizeUnit: 'px',
  css: ''
};

const Component = onlyUpdateForKeys(['loading', 'color', 'size', 'sizeUnit', 'css'])(Loader);
Component.defaultProps = Loader.defaultProps;
export default Component;
