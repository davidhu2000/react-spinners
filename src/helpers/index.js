export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
