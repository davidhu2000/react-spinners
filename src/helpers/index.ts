export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
