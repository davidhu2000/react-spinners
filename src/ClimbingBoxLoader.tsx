import * as React from "react";

import { cssValue } from "./helpers";
import { LoaderSizeProps } from "./helpers/props";
import { createAnimation } from "./helpers/animation";

const climbingBox = createAnimation(
  "ClimbingBoxLoader",
  `0% {transform:translate(0, -1em) rotate(-45deg)}
  5% {transform:translate(0, -1em) rotate(-50deg)}
  20% {transform:translate(1em, -2em) rotate(47deg)}
  25% {transform:translate(1em, -2em) rotate(45deg)}
  30% {transform:translate(1em, -2em) rotate(40deg)}
  45% {transform:translate(2em, -3em) rotate(137deg)}
  50% {transform:translate(2em, -3em) rotate(135deg)}
  55% {transform:translate(2em, -3em) rotate(130deg)}
  70% {transform:translate(3em, -4em) rotate(217deg)}
  75% {transform:translate(3em, -4em) rotate(220deg)}
  100% {transform:translate(0, -1em) rotate(-225deg)}`,
  "climbingBox"
);

function ClimbingBoxLoader({
  loading = true,
  color = "#000000",
  speedMultiplier = 1,
  css = {},
  size = 15,
  ...additionalprops
}: LoaderSizeProps): JSX.Element | null {
  const style: React.CSSProperties = {
    position: "absolute",
    left: "0",
    bottom: "-0.1em",
    height: "1em",
    width: "1em",
    backgroundColor: "transparent",
    borderRadius: "15%",
    border: `0.25em solid ${color}`,
    transform: "translate(0, -1em) rotate(-45deg)",
    animationFillMode: "both",
    animation: `${climbingBox} ${2.5 / speedMultiplier}s infinite cubic-bezier(0.79, 0, 0.47, 0.97)`,
  };

  const wrapper: React.CSSProperties = {
    position: "absolute",
    top: "50%",
    left: "50%",
    marginTop: "-2.7em",
    marginLeft: "-2.7em",
    width: "5.4em",
    height: "5.4em",
    fontSize: cssValue(size),
  };

  const hill: React.CSSProperties = {
    position: "absolute",
    width: "7.1em",
    height: "7.1em",
    top: "1.7em",
    left: "1.7em",
    borderLeft: `0.25em solid ${color}`,
    transform: "rotate(45deg)",
  };

  const container: React.CSSProperties = {
    position: "relative",
    width: "7.1em",
    height: "7.1em",
    ...css,
  };

  if (!loading) {
    return null;
  }

  return (
    <span style={container} {...additionalprops}>
      <span style={wrapper}>
        <span style={style} />
        <span style={hill} />
      </span>
    </span>
  );
}

export default ClimbingBoxLoader;
