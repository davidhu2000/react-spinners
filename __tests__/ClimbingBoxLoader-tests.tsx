import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from "jest-emotion";
expect.extend(matchers);

import ClimbingBoxLoader from "../src/ClimbingBoxLoader";
import { LoaderSizeProps } from "../src/interfaces";
import { sizeDefaults } from "../src/helpers";

describe("ClimbingBoxLoader", () => {
  let loader: ReactWrapper;
  let props: LoaderSizeProps;
  let defaultColor: string = "#000000";
  let defaultSize: number = 15;
  let defaultUnit: string = "px";

  it("should match snapshot", () => {
    loader = mount(<ClimbingBoxLoader />);
    expect(loader).toMatchSnapshot();
  });

  it("should contain default props if no props are passed", () => {
    props = loader.props();
    expect(props).toEqual(sizeDefaults(defaultSize));
  });

  it("parent div should contain styles created using default props", () => {
    expect(loader.find("div div")).toHaveStyleRule("font-size", `${defaultSize}${defaultUnit}`);
  });

  it("children div should contain styles created using default props", () => {
    expect(loader.find("div div div").at(0)).toHaveStyleRule(
      "border",
      `0.25em solid ${defaultColor}`
    );
    expect(loader.find("div div div").at(1)).toHaveStyleRule(
      "border-left",
      `0.25em solid ${defaultColor}`
    );
  });

  it("should render null if loading prop is set as false", () => {
    loader = mount(<ClimbingBoxLoader loading={false} />);
    expect(loader.isEmptyRender()).toBe(true);
  });

  it("should render the correct color based on prop", () => {
    let color: string = "#e2e2e2";
    loader = mount(<ClimbingBoxLoader color={color} />);

    expect(loader.find("div div div").at(0)).not.toHaveStyleRule(
      "border",
      `0.25em solid ${defaultColor}`
    );
    expect(loader.find("div div div").at(0)).toHaveStyleRule("border", `0.25em solid ${color}`);

    expect(loader.find("div div div").at(1)).not.toHaveStyleRule(
      "border-left",
      `0.25em solid ${defaultColor}`
    );
    expect(loader.find("div div div").at(1)).toHaveStyleRule(
      "border-left",
      `0.25em solid ${color}`
    );
  });

  it("should render the correct size based on props", () => {
    let size: number = 18;
    loader = mount(<ClimbingBoxLoader size={size} />);
    expect(loader.find("div div")).not.toHaveStyleRule("font-size", `${defaultSize}${defaultUnit}`);
    expect(loader.find("div div")).toHaveStyleRule("font-size", `${size}${defaultUnit}`);
  });

  it("should render the correct sizeUnit based on props", () => {
    let unit: string = "%";
    loader = mount(<ClimbingBoxLoader sizeUnit={unit} />);
    expect(loader.find("div div")).not.toHaveStyleRule("font-size", `${defaultSize}${defaultUnit}`);
    expect(loader.find("div div")).toHaveStyleRule("font-size", `${defaultSize}${unit}`);
  });

  it("should render the css override based on props", () => {
    loader = mount(
      <ClimbingBoxLoader css={"position: absolute; width: 100px; height: 200px; color: blue;"} />
    );
    expect(loader).not.toHaveStyleRule("position", "relative");
    expect(loader).not.toHaveStyleRule("width", "7.1em");
    expect(loader).not.toHaveStyleRule("height", "7.1em");
    expect(loader).toHaveStyleRule("position", "absolute");
    expect(loader).toHaveStyleRule("width", "100px");
    expect(loader).toHaveStyleRule("height", "200px");
    expect(loader).toHaveStyleRule("color", "blue");
  });
});
