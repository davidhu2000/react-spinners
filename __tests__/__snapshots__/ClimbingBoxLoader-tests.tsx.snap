// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClimbingBoxLoader should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translate(0, -1em) rotate(-45deg);
    -moz-transform: translate(0, -1em) rotate(-45deg);
    -ms-transform: translate(0, -1em) rotate(-45deg);
    transform: translate(0, -1em) rotate(-45deg);
  }

  5% {
    -webkit-transform: translate(0, -1em) rotate(-50deg);
    -moz-transform: translate(0, -1em) rotate(-50deg);
    -ms-transform: translate(0, -1em) rotate(-50deg);
    transform: translate(0, -1em) rotate(-50deg);
  }

  20% {
    -webkit-transform: translate(1em, -2em) rotate(47deg);
    -moz-transform: translate(1em, -2em) rotate(47deg);
    -ms-transform: translate(1em, -2em) rotate(47deg);
    transform: translate(1em, -2em) rotate(47deg);
  }

  25% {
    -webkit-transform: translate(1em, -2em) rotate(45deg);
    -moz-transform: translate(1em, -2em) rotate(45deg);
    -ms-transform: translate(1em, -2em) rotate(45deg);
    transform: translate(1em, -2em) rotate(45deg);
  }

  30% {
    -webkit-transform: translate(1em, -2em) rotate(40deg);
    -moz-transform: translate(1em, -2em) rotate(40deg);
    -ms-transform: translate(1em, -2em) rotate(40deg);
    transform: translate(1em, -2em) rotate(40deg);
  }

  45% {
    -webkit-transform: translate(2em, -3em) rotate(137deg);
    -moz-transform: translate(2em, -3em) rotate(137deg);
    -ms-transform: translate(2em, -3em) rotate(137deg);
    transform: translate(2em, -3em) rotate(137deg);
  }

  50% {
    -webkit-transform: translate(2em, -3em) rotate(135deg);
    -moz-transform: translate(2em, -3em) rotate(135deg);
    -ms-transform: translate(2em, -3em) rotate(135deg);
    transform: translate(2em, -3em) rotate(135deg);
  }

  55% {
    -webkit-transform: translate(2em, -3em) rotate(130deg);
    -moz-transform: translate(2em, -3em) rotate(130deg);
    -ms-transform: translate(2em, -3em) rotate(130deg);
    transform: translate(2em, -3em) rotate(130deg);
  }

  70% {
    -webkit-transform: translate(3em, -4em) rotate(217deg);
    -moz-transform: translate(3em, -4em) rotate(217deg);
    -ms-transform: translate(3em, -4em) rotate(217deg);
    transform: translate(3em, -4em) rotate(217deg);
  }

  75% {
    -webkit-transform: translate(3em, -4em) rotate(220deg);
    -moz-transform: translate(3em, -4em) rotate(220deg);
    -ms-transform: translate(3em, -4em) rotate(220deg);
    transform: translate(3em, -4em) rotate(220deg);
  }

  100% {
    -webkit-transform: translate(0, -1em) rotate(-225deg);
    -moz-transform: translate(0, -1em) rotate(-225deg);
    -ms-transform: translate(0, -1em) rotate(-225deg);
    transform: translate(0, -1em) rotate(-225deg);
  }
}

.emotion-0fpqs8j {
  position: relative;
  width: 7.1em;
  height: 7.1em;
}

.emotion-019shb47 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -2.7em;
  margin-left: -2.7em;
  width: 5.4em;
  height: 5.4em;
  font-size: 15px;
}

.emotion-0bujqyr {
  position: absolute;
  left: 0;
  bottom: -0.1em;
  height: 1em;
  width: 1em;
  background-color: transparent;
  border-radius: 15%;
  border: 0.25em solid #000000;
  -webkit-transform: translate(0, -1em) rotate(-45deg);
  -moz-transform: translate(0, -1em) rotate(-45deg);
  -ms-transform: translate(0, -1em) rotate(-45deg);
  transform: translate(0, -1em) rotate(-45deg);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 2.5s infinite cubic-bezier(0.79, 0, 0.47, 0.97);
  animation: animation-0 2.5s infinite cubic-bezier(0.79, 0, 0.47, 0.97);
}

.emotion-0o4f6ni {
  position: absolute;
  width: 7.1em;
  height: 7.1em;
  top: 1.7em;
  left: 1.7em;
  border-left: 0.25em solid #000000;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<Loader
  color="#000000"
  css=""
  loading={true}
  size={15}
>
  <span
    className="emotion-0"
  >
    <span
      className="emotion-0fpqs8j"
    >
      <span
        className="emotion-019shb47"
      >
        <EmotionCssPropInternal
          __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
          css={
            Object {
              "map": undefined,
              "name": "bujqyr",
              "next": Object {
                "name": "animation-dl5pip",
                "next": undefined,
                "styles": "@keyframes animation-dl5pip{
  0% {transform:translate(0, -1em) rotate(-45deg)}
  5% {transform:translate(0, -1em) rotate(-50deg)}
  20% {transform:translate(1em, -2em) rotate(47deg)}
  25% {transform:translate(1em, -2em) rotate(45deg)}
  30% {transform:translate(1em, -2em) rotate(40deg)}
  45% {transform:translate(2em, -3em) rotate(137deg)}
  50% {transform:translate(2em, -3em) rotate(135deg)}
  55% {transform:translate(2em, -3em) rotate(130deg)}
  70% {transform:translate(3em, -4em) rotate(217deg)}
  75% {transform:translate(3em, -4em) rotate(220deg)}
  100% {transform:translate(0, -1em) rotate(-225deg)}
}",
              },
              "styles": "
      position: absolute;
      left: 0;
      bottom: -0.1em;
      height: 1em;
      width: 1em;
      background-color: transparent;
      border-radius: 15%;
      border: 0.25em solid #000000;
      transform: translate(0, -1em) rotate(-45deg);
      animation-fill-mode: both;
      animation: animation-dl5pip 2.5s infinite cubic-bezier(0.79, 0, 0.47, 0.97);
    ",
              "toString": [Function],
            }
          }
        >
          <span
            className="emotion-0bujqyr"
          />
        </EmotionCssPropInternal>
        <EmotionCssPropInternal
          __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
          css={
            Object {
              "map": undefined,
              "name": "o4f6ni",
              "next": undefined,
              "styles": "
      position: absolute;
      width: 7.1em;
      height: 7.1em;
      top: 1.7em;
      left: 1.7em;
      border-left: 0.25em solid #000000;
      transform: rotate(45deg);
    ",
              "toString": [Function],
            }
          }
        >
          <span
            className="emotion-0o4f6ni"
          />
        </EmotionCssPropInternal>
      </span>
    </span>
  </span>
</Loader>
`;
